### DISCUSSION TITLE
"Tile Z/X/Y Tile is outside image bounds" at Certain Zoom Levels

### DISCUSSION THREAD

**Discussion by aemonm** [developmentseed/titiler#1064] [General]
We're noticing tile requests are returning 404s with the error "Tile Z/X/Y Tile is outside image bounds" when there should be tile data returned. This happens at certain zoom levels. 

See these [two](https://cerulean.skytruth.org/?startDate=2024-12-27&endDate=2024-12-27&cls=2_3_4_5_6_7_8&sources=1_2_3&sourceScore=-Infinity_Infinity&sourceLimit=3&machine_confidence=0.5&area=0_Infinity&mpa=false&eez=false&iho=false&hitl_cls=&offset=0&limit=100&hitl_rev=0_1&zoom=9.895348&lat=4.378028&lng=-3.141774&slickId=3225739) [examples](https://cerulean.skytruth.org/?startDate=2000-06-06&endDate=2024-12-07&hitl_cls=&area=0_Infinity&machine_confidence=0.5&sourceScore=Infinity_Infinity&zoom=9.696582&lat=23.539547&lng=60.793135&slickId=3149697) where the imagery is clipped at the bottom left (you can zoom in/out to see the full imagery appear). 

Here's a screenshot showing that tiles from the bottom row aren't appearing and their corresponding requests are 404s with the above error message.
![image](https://github.com/user-attachments/assets/65ec0e01-2cd2-4b00-acda-5369d5d4f831)
![image](https://github.com/user-attachments/assets/2f21f344-5cd5-423f-b64e-ab682d381b97)

We are running an old version of TiTiler and could upgrade if this was a known issue that's been fixed in recent versions (couldn't determine if that's the case from the changelog).

TiTiler Version: 0.5.1

**Comment by vincentsarago** [comment]
```python
with rasterio.Env(AWS_REQUEST_PAYER="requester"):
    with S1L1CReader("S1A_IW_GRDH_1SDV_20230602T140013_20230602T140042_048808_05DE99_BDCF") as s1:
        print(s1.bounds)
        print(s1.tile_exists(10942, 7092, 14))
    with Reader(s1._get_band_url("vv")) as src:
        print(src.bounds)
        print(src.tile_exists(10942, 7092, 14))


(60.00380832650571, 23.585737344738313, 62.77578173537508, 25.627327750202888)
False

(60.00179037392455, 23.462311304476383, 62.85615656898099, 25.63674637611014)
True
```

This is tricky and could be considered as a rio-tiler/rio-tiler-pds issue. 

The problem is that in rio-tiler-pds, we used the bounds derived from the metadata `.json` https://github.com/cogeotiff/rio-tiler-pds/blob/ee68026425fcace7d1ae85e77dcaa541f8e06931/rio_tiler_pds/sentinel/aws/sentinel1.py#L91-L92

but for some reason the bounds DO NOT match the dataset bounds 😬, resulting in `tile_exists()` check returning False instead of `True` 


I'm not quite sure how to solve this right now :-(

**Reply by aemonm** [reply]
@vincentsarago we've come across what looks to be another [example of this](https://cerulean.skytruth.org/?startDate=2025-01-29&endDate=2025-01-29&cls=2_3_4_5_6_7_8&sources=1_2_3&sourceScore=-Infinity_Infinity&sourceLimit=3&machine_confidence=0.5&area=0_Infinity&mpa=false&eez=false&iho=false&hitl_cls=&offset=0&limit=100&hitl_rev=0_1&zoom=9.57135&lat=36.464198&lng=35.494692&slickId=3808659).

Do you recommend raising this as an issue with rio-tiler/rio-tiler-pds?

**Reply by vincentsarago** [reply]
The main issue is that the bounds derived from the `productInfo.json` is wrong or maybe we are missing something 

you could overcome this by creating your own custom Reader 


```python
import attr
from rio_tiler_pds.sentinel.aws import S1L1CReader

@attr.s
class CustomReader(S1L1CReader):

    def tile_exists(self, tile_x: int, tile_y: int, tile_z: int) -> bool:
        return True
```

This will avoid the first `tile_exits`. There will be `tile_exits` check for each band later in the process
